catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -81,9 +84,110 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
+    private final CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction;
+
+    private final CheckedFunction<PDDocument, String> geoParserFunction;
+
+    private final CheckedFunction<PDDocument, Optional<byte[]>> thumbnailFunction;
+
     private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
             Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
 
+    private MetacardType metacardType;
+
+    private boolean usePdfTitleAsTitle;
+
+    /**
+     * @param metacardType       must be non-null
+     * @param usePdfTitleAsTitle must be non-null
+     */
+    public PdfInputTransformer(MetacardType metacardType, Boolean usePdfTitleAsTitle) {
+        this(metacardType,
+                usePdfTitleAsTitle,
+                PDDocument::load,
+                GEO_PDF_PARSER::getWktFromPDF,
+                PdfInputTransformer::generatePdfThumbnail);
```

Comment:
@glenhein Ignore my comment. I see that it's been updated to use the function on line 258. I'm still not in love with the testing-specific code smell. It does make this class more complicated than it otherwise needs to be for no obvious reason without the context of the unit tests. It might be cleaner if you could extract the logic for the three replaceable operations to another class, inject a bean for that in your blueprint, and mock that class and its three methods in your unit test.

It removes the smell but does lead to some extra complexity. However, from the perspective of someone looking at your class (and this new bean) in a vacuum it won't look strange.